### PR TITLE
feat: Handle territory timers in a more advanced way to make sure they always appear as accurate as they can

### PR DESCRIPTION
**Pull request by kristofbolyai**

### CODE REVIEW

**Review by magicus — APPROVED** [review]

File: common/src/main/java/com/wynntils/models/territories/GuildAttackTimerModel.java [hunk 1]
```diff
@@ -13,30 +13,51 @@
...
+ * - Use chat messages when we can, as they are more reliable
+ * - Use scoreboard messages when we don't have a chat message
+ * - If a chat message is received for a territory, we stop using the scoreboard for that territory
+ * - We also take territory defenses from chat messages
+ */
 public final class GuildAttackTimerModel extends Model {
     private static final Pattern GUILD_ATTACK_PATTERN = Pattern.compile("§b- (.+):(.+) §3(.+)");
     private static final Pattern GUILD_DEFENSE_CHAT_PATTERN = Pattern.compile("§3.+§b (.+) defense is (.+)");
+    private static final Pattern WAR_MESSAGE_PATTERN = Pattern.compile(
+            "§3\\[WAR\\]§c The war for (?<guild>.+) will start in (?<remaining>.+) (?<type>minutes|seconds)\\.");
```
Comment (ShadowCat117) [inline common/src/main/java/com/wynntils/models/territories/GuildAttackTimerModel.java]: Minor nitpick but the guild group should be called territory

**Comment by ShadowCat117** [comment]
Code looks good, can't test it but I don't see why it wouldn't work. Only exception is I think you need to handle refunds to remove timers

**Comment by kristofbolyai** [comment]
> Code looks good, can't test it but I don't see why it wouldn't work. Only exception is I think you need to handle refunds to remove timers

Refunds are handled.

File: common/src/main/java/com/wynntils/models/territories/GuildAttackTimerModel.java (hunk 1 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/territories/GuildAttackTimerModel.java]: Thanks!

**Comment by ShadowCat117** [comment]
> Refunds are handled.

Oh I see, the captured message appears if a different guild takes the territory so no need to check for the refund message

**Comment by kristofbolyai** [comment]
> > Refunds are handled.
> 
> Oh I see, the captured message appears if a different guild takes the territory so no need to check for the refund message

Yeah. At least I really think so, because all of my logs have a captured message right before the refund message...

**Pull request merged**